### ISSUE TITLE
Allow pass custom class name to `@icon` annotation

### ISSUE THREAD

**Issue by Chaosus** [godotengine/godot-proposals#9972]
### Describe the project you are working on

Godot

### Describe the problem or limitation you are having in your project

This is general editor enchancement.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

The Godot has a rich collection of embedded icons - why not allow using them for other user classes? Sometimes It's tedious to invent a proper new icon, or existed one fits better.

The idea is to make icon annotation to accept class name in addition to icon path. eg:

`@icon(AnimatedSprite)`

And then the custom user class will use the icon originally intended for `AnimatedSprite`.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

^^

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Nope

### Is there a reason why this should be core and not an add-on in the asset library?

N/A

**Comment by KoBeWi** [comment]
You can make a custom Texture2D sub-class that holds editor texture and set one as icon.

**Comment by Calinou** [comment]
I'm not sure if we should make it easier to use existing node icons as icons for custom nodes. This is bound to cause a lot of confusion if you use another node's icon if it's not the icon of the immediate parent node.

**Comment by Mickeon** [comment]
Personally, I would rather _somehow_ allow all icons that the Editor theme has to offer to be used as icons. That way, a conscious choice can be made to avoid existing Node icons if desired, while still offering a great selection of existing icons.

**Comment by BaconEggsRL** [comment]
I agree it would be nice to use the Editor icons, even just as a temporary icon while testing.
There are already conventient ways to browse and download the editor icons:
https://godotengine.github.io/editor-icons/
https://github.com/godotengine/godot/tree/master/editor/icons
And we can use
```
editor_theme = EditorInterface.get_editor_theme()
icon = editor_theme.get_icon(icon_name, EDITOR_ICONS)
```
to retrieve editor icons already, but it doesn't work for the ```@icon``` tag, since we must supply a string literal.

This would just circumvent the need to download the icon if one wanted to use an existing icon.

**Comment by KoBeWi** [comment]
Updated code from my last comment:
```GDScript
@tool
class_name EditorTexture extends ImageTexture

static var icon_cache: String

@export var icon_name: StringName:
	set(i):
		icon_name = i
		
		if Engine.is_editor_hint():
			if EditorInterface.get_editor_theme().has_icon(icon_name, &"EditorIcons"):
				var texture := EditorInterface.get_editor_theme().get_icon(icon_name, &"EditorIcons")
				set_image(texture.get_image())

func _validate_property(property: Dictionary) -> void:
	if not Engine.is_editor_hint():
		return
	
	if property["name"] == "image":
		property["usage"] = PROPERTY_USAGE_NONE
	elif property["name"] == "icon_name":
		if icon_cache.is_empty():
			icon_cache = ",".join(EditorInterface.get_editor_theme().get_icon_list(&"EditorIcons"))
				
			
		property["hint"] = PROPERTY_HINT_ENUM_SUGGESTION
		property["hint_string"] = icon_cache
```
Example usage:

https://github.com/user-attachments/assets/c5b908cb-67b1-44b9-9525-32ab9096bf17
